lib/index.js

```diff
@@ -128,6 +128,14 @@ internals.authenticate = async function(token, options, request, h) {
   try {
     decoded = JWT.decode(token, { complete: options.complete || false });
   } catch (e) {
+    // fix for https://github.com/dwyl/hapi-auth-jwt2/issues/328 -
+    // JWT.decode() can fail either by throwing an exception or by
+    // returning null, so here we just fall through to the following
+    // block that tests if decoded is not set, so that we can handle
+    // both failure types at once
+  }
+
+  if (!decoded) {
```

Comment:
OK, so all we are doing here is returning the error Object if `decoded` remains unset/undefined after attempted `JWT.decode` as opposed to returning the error in the `catch` block. 💭  Makes sense. 👍